cogs/general/general.py

```diff
@@ -201,16 +214,14 @@ def __init__(self, bot):
 
         game = "Chilling in {} status".format(user.status)
 
-        if user.game is None:
-            pass
-        elif user.game.url is None:
-            game = "Playing {}".format(user.game)
-        else:
+        if user.game and user.game.name and user.game.url:
             game = "Streaming: [{}]({})".format(user.game, user.game.url)
+        elif user.game and user.game.name:
+            game = "Playing {}".format(user.game)
+
 
         if roles:
-            roles = sorted(roles, key=[x.name for x in guild.role_hierarchy
-                                       if x.name != "@everyone"].index)
+            roles = sorted(user.roles)
```

Comment:
You probably want to just move this line to the top a bit.

First initialise:

`roles = sorted(user.roles)[1:]`

Then check if it exists and transform it:

```py
if roles:
  roles = ', '.join(x.name for x in roles)
```